src/loader.js

```diff
@@ -0,0 +1,38 @@
+let isLoadingLibrary = false
+const callbacks = []
```

Comment:
some "why" docs on this approach would be helpful.

Reply:
Okay, added